Jax upgrade errors

- Error on dualconv2d

**Logs:**

```
============================================================================ test session starts ============================================================================
platform linux -- Python 3.10.12, pytest-6.2.4, py-1.11.0, pluggy-1.0.0.dev0 -- /opt/ttforge-toolchain/venv/bin/python
cachedir: .pytest_cache
rootdir: /tt-forge-fe, configfile: pytest.ini
plugins: split-0.10.0, forked-1.6.0, anyio-4.9.0, timeout-2.0.1, xdist-2.5.0
collected 1 item                                                                                                                                                            

forge/test/mlir/test_ops_tf.py::test_dual_conv2d FAILED                                                                                                               [100%]

================================================================================= FAILURES ==================================================================================
_____________________________________________________________________________ test_dual_conv2d ______________________________________________________________________________

    @pytest.mark.push
    def test_dual_conv2d():
    
        tf.random.set_seed(0)
    
        class DualConv2d(tf.keras.Model):
            def __init__(self):
                super().__init__()
                self.conv2d1 = tf.keras.layers.Conv2D(
                    64, (3, 3), strides=[1, 1], padding="same", data_format="channels_last"
                )
                self.conv2d2 = tf.keras.layers.Conv2D(
                    256, (2, 2), strides=[2, 2], padding="valid", data_format="channels_last"
                )
    
            def call(self, x):
                return self.conv2d2(self.conv2d1(x))
    
        inputs = [tf.random.uniform((1, 128, 128, 3))]
    
        framework_model = DualConv2d()
>       compiled_model = forge.compile(framework_model, sample_inputs=inputs)

forge/test/mlir/test_ops_tf.py:119: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
forge/forge/compile.py:255: in compile_main
    return forge_compile_from_context(compile_context)
forge/forge/compile.py:297: in forge_compile_from_context
    next_stage = stage_to_func[current_stage](context)
forge/forge/compile.py:686: in generate_initial_graph
    context.graph, context.outputs, context.intermediate_tensors, context.inputs, _ = generate_graph(
forge/forge/compile.py:1141: in generate_graph
    outputs = module.forward(*outputs)
forge/forge/module.py:669: in wrap_forward
    return orig_forward(*args, **kwargs)
generated_modules/DualConv2d.py:21: in forward
    transpose_3 = forge.op.Transpose("", self.get_parameter("kernel"), dim0=-4, dim1=-1)
forge/forge/op/tm.py:44: in Transpose
    return op("transpose", name, operandA, attrs=(dim0, dim1), dim0=dim0, dim1=dim1).get_tensor()
forge/forge/op/common.py:72: in get_tensor
    result.set_value(ref_output)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Forge Empty Tensor: (64, 3, 3, 3)
value = tensor([[[[-0.0243,  0.0285],
          [ 0.0579, -0.0170],
          [-0.0615, -0.0203],
          ...,
          [-0...       [-0.0138, -0.0342],
          [ 0.0536,  0.0360],
          [-0.0606,  0.0605]]]], grad_fn=<TransposeBackward0>)

    def set_value(self, value: torch.Tensor):
        assert (
>           self.tensor_shape.get_pytorch_shape() == value.shape
        ), f"Setting a tensor value of incorrect shape: {self.tensor_shape.get_pytorch_shape()} vs {value.shape}"
E       AssertionError: Setting a tensor value of incorrect shape: (64, 3, 3, 3) vs torch.Size([256, 2, 64, 2])

forge/forge/tensor.py:420: AssertionError
--------------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------------
2025-07-11 08:41:38.283259: E external/local_xla/xla/stream_executor/cuda/cuda_platform.cc:51] failed call to cuInit: INTERNAL: CUDA error: Failed call to cuInit: UNKNOWN ERROR (303)
2025-07-11 08:41:38.293 | INFO     | forge.compile:compile_main:228 - Compiling module DualConv2d
2025-07-11 08:41:38.294 | INFO     | forge.compile:forge_compile_from_context:290 - Running compile stage init_compile
2025-07-11 08:41:38.294 | INFO     | forge.compile:forge_compile_from_context:290 - Running compile stage generate_initial_graph
I0000 00:00:1752223298.712496  134648 devices.cc:67] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1752223298.712603  134648 single_machine.cc:374] Starting new session
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'kernel' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'bias' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'kernel' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'transpose_3' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'transpose_4' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'inputs_0' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'transpose_5' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'bias' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'reshape_7' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'squeeze_8' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'squeeze_9' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'conv2d_6' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'squeeze_10' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'const_00' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'const_10' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'add_11' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'const_00' does not have a framework dtype specified. Using TVM generated dtype.
2025-07-11 08:41:40.207 | DEBUG    | forge.tvm_to_python:_determine_node_dtype:2197 - Node 'const_10' does not have a framework dtype specified. Using TVM generated dtype.
------------------------------------------------------------------------- Captured stderr teardown --------------------------------------------------------------------------
2025-07-11 08:41:40.393 | INFO     | conftest:memory_usage_tracker:140 - Test memory usage:
2025-07-11 08:41:40.393 | INFO     | conftest:memory_usage_tracker:141 -     By test: 294.08 MB
2025-07-11 08:41:40.393 | INFO     | conftest:memory_usage_tracker:142 -     Minimum: 1278.61 MB
2025-07-11 08:41:40.393 | INFO     | conftest:memory_usage_tracker:143 -     Maximum: 1572.69 MB
2025-07-11 08:41:40.393 | INFO     | conftest:memory_usage_tracker:144 -     Average: 1480.90 MB
2025-07-11 08:41:40.544 | INFO     | conftest:memory_usage_tracker:148 - Memory usage after garbage collection: 1572.69 MB
2025-07-11 08:41:40.544 | INFO     | conftest:memory_usage_tracker:152 - Memory usage after malloc_trim: 1563.39 MB
=========================================================================== slowest 10 durations ============================================================================
1.93s call     forge/test/mlir/test_ops_tf.py::test_dual_conv2d
0.24s teardown forge/test/mlir/test_ops_tf.py::test_dual_conv2d
0.01s setup    forge/test/mlir/test_ops_tf.py::test_dual_conv2d
========================================================================== short test summary info ==========================================================================
FAILED forge/test/mlir/test_ops_tf.py::test_dual_conv2d - AssertionError: Setting a tensor value of incorrect shape: (64, 3, 3, 3) vs torch.Size([256, 2, 64, 2])
============================================================================= 1 failed in 2.32s ======================================================================
```

- RuntimeError in test_resnet_tensorflow:

**RuntimeError: Dimension out of range (expected to be in range of [-2, 1], but got -4)**

```
    @pytest.mark.push
    @pytest.mark.nightly
    def test_resnet_tensorflow():
    
        # Record model details
        module_name = record_model_properties(
            framework=Framework.TENSORFLOW,
            model=ModelArch.RESNET,
            variant="resnet50",
            source=Source.KERAS,
            task=Task.IMAGE_CLASSIFICATION,
        )
    
        # Load Resnet50 Model
        framework_model = ResNet50(weights="imagenet")
    
        # Load sample inputs
        sample_input = get_sample_inputs()
        inputs = [sample_input]
    
        # Compile model
>       compiled_model = forge.compile(framework_model, inputs, module_name=module_name)

forge/test/models/tensorflow/vision/resnet/test_resnet.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/compile.py:255: in compile_main
    return forge_compile_from_context(compile_context)
/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/compile.py:297: in forge_compile_from_context
    next_stage = stage_to_func[current_stage](context)
/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/compile.py:686: in generate_initial_graph
    context.graph, context.outputs, context.intermediate_tensors, context.inputs, _ = generate_graph(
/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/compile.py:1141: in generate_graph
    outputs = module.forward(*outputs)
/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/module.py:669: in wrap_forward
    return orig_forward(*args, **kwargs)
generated_modules/tf_resnet_resnet50_img_cls_keras_538.py:234: in forward
    transpose_7 = forge.op.Transpose("", self.get_parameter("kernel"), dim0=-4, dim1=-1)
/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/op/tm.py:44: in Transpose
    return op("transpose", name, operandA, attrs=(dim0, dim1), dim0=dim0, dim1=dim1).get_tensor()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <forge.op.common.ForgeOp object at 0x7f603ab8ac80>, out_df = None

    def get_tensor(self, out_df=None) -> Tensor:
        """
        Generate result tensor of the right shape, and if value is set, value.
        """
    
        # get reference output shape
        shapes = [o.shape.get_pytorch_shape() for o in self.operands]
        shape, self.operand_broadcast = self.cpp_op_type.shape(shapes)
    
        # get reference output value
        values = [o.value() if isinstance(o, (Tensor, Parameter)) else o for o in self.operands]
>       ref_output = self.cpp_op_type.eval(values)
E       RuntimeError: Dimension out of range (expected to be in range of [-2, 1], but got -4)
E       Exception raised from maybe_wrap_dim_slow at /pytorch/c10/core/WrapDimMinimal.cpp:23 (most recent call first):
E       frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x98 (0x7f6320347218 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libc10.so)
E       frame #1: long c10::detail::maybe_wrap_dim_slow<long>(long, long, bool) + 0x27b (0x7f632032c9eb in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libc10.so)
E       frame #2: at::native::transpose(at::Tensor const&, long, long) + 0x50 (0x7f63221f6290 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #3: <unknown function> + 0x2c03523 (0x7f6322fbf523 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #4: at::_ops::transpose_int::redispatch(c10::DispatchKeySet, at::Tensor const&, long, long) + 0x7d (0x7f6322b2439d in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #5: <unknown function> + 0x50f580a (0x7f63254b180a in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #6: <unknown function> + 0x50f5aa6 (0x7f63254b1aa6 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #7: at::_ops::transpose_int::redispatch(c10::DispatchKeySet, at::Tensor const&, long, long) + 0x7d (0x7f6322b2439d in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #8: <unknown function> + 0x4b6f2df (0x7f6324f2b2df in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #9: <unknown function> + 0x4b6f776 (0x7f6324f2b776 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #10: at::_ops::transpose_int::call(at::Tensor const&, long, long) + 0x196 (0x7f6322b95076 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
E       frame #11: tt::ops::transpose::eval(tt::ops::Op const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&) + 0xc5 (0x7f62bcc6ef45 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/_C.so)
E       frame #12: tt::ops::Op::eval(tt::graphlib::OpType const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&) const + 0x80 (0x7f62bcc5b8c0 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/_C.so)
E       frame #13: tt::graphlib::OpType::eval(std::vector<at::Tensor, std::allocator<at::Tensor> > const&) const + 0x13 (0x7f62bcb79cb3 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/_C.so)
E       frame #14: <unknown function> + 0x29759a (0x7f62bcbcb59a in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/_C.so)
E       frame #15: <unknown function> + 0x29750e (0x7f62bcbcb50e in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/_C.so)
E       frame #16: <unknown function> + 0xf5845 (0x7f62bca29845 in /opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/_C.so)
E       frame #17: <unknown function> + 0x18ae52 (0x5635df9ade52 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #18: _PyObject_MakeTpCall + 0x25b (0x5635df9a477b in /opt/ttforge-toolchain/venv/bin/python)
E       frame #19: <unknown function> + 0x198bfb (0x5635df9bbbfb in /opt/ttforge-toolchain/venv/bin/python)
E       frame #20: _PyEval_EvalFrameDefault + 0x5633 (0x5635df99d8a3 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #21: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #22: _PyEval_EvalFrameDefault + 0x807 (0x5635df998a77 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #23: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #24: _PyEval_EvalFrameDefault + 0x1987 (0x5635df999bf7 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #25: <unknown function> + 0x198a6e (0x5635df9bba6e in /opt/ttforge-toolchain/venv/bin/python)
E       frame #26: _PyEval_EvalFrameDefault + 0x2a7b (0x5635df99aceb in /opt/ttforge-toolchain/venv/bin/python)
E       frame #27: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #28: _PyEval_EvalFrameDefault + 0x2a7b (0x5635df99aceb in /opt/ttforge-toolchain/venv/bin/python)
E       frame #29: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #30: _PyEval_EvalFrameDefault + 0x1987 (0x5635df999bf7 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #31: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #32: _PyEval_EvalFrameDefault + 0x6c0 (0x5635df998930 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #33: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #34: _PyEval_EvalFrameDefault + 0x6c0 (0x5635df998930 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #35: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #36: _PyEval_EvalFrameDefault + 0x1987 (0x5635df999bf7 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #37: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #38: _PyEval_EvalFrameDefault + 0x2a7b (0x5635df99aceb in /opt/ttforge-toolchain/venv/bin/python)
E       frame #39: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #40: _PyEval_EvalFrameDefault + 0x2a7b (0x5635df99aceb in /opt/ttforge-toolchain/venv/bin/python)
E       frame #41: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #42: _PyEval_EvalFrameDefault + 0x1987 (0x5635df999bf7 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #43: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #44: _PyEval_EvalFrameDefault + 0x5633 (0x5635df99d8a3 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #45: <unknown function> + 0x198921 (0x5635df9bb921 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #46: _PyEval_EvalFrameDefault + 0x5633 (0x5635df99d8a3 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #47: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #48: _PyObject_FastCallDictTstate + 0x16d (0x5635df9a39fd in /opt/ttforge-toolchain/venv/bin/python)
E       frame #49: _PyObject_Call_Prepend + 0x5c (0x5635df9b8a2c in /opt/ttforge-toolchain/venv/bin/python)
E       frame #50: <unknown function> + 0x29d854 (0x5635dfac0854 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #51: _PyObject_MakeTpCall + 0x25b (0x5635df9a477b in /opt/ttforge-toolchain/venv/bin/python)
E       frame #52: _PyEval_EvalFrameDefault + 0x6907 (0x5635df99eb77 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #53: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #54: _PyEval_EvalFrameDefault + 0x807 (0x5635df998a77 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #55: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #56: _PyEval_EvalFrameDefault + 0x2a7b (0x5635df99aceb in /opt/ttforge-toolchain/venv/bin/python)
E       frame #57: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #58: _PyEval_EvalFrameDefault + 0x1987 (0x5635df999bf7 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #59: _PyFunction_Vectorcall + 0x7c (0x5635df9ae6ac in /opt/ttforge-toolchain/venv/bin/python)
E       frame #60: _PyEval_EvalFrameDefault + 0x5633 (0x5635df99d8a3 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #61: <unknown function> + 0x198921 (0x5635df9bb921 in /opt/ttforge-toolchain/venv/bin/python)
E       frame #62: _PyEval_EvalFrameDefault + 0x5633 (0x5635df99d8a3 in /opt/ttforge-toolchain/venv/bin/python)

/opt/ttforge-toolchain/venv/lib/python3.10/site-packages/forge/op/common.py:61: RuntimeError
```

_Originally posted by @saiarthiraguram in https://github.com/tenstorrent/tt-forge-fe/issues/2346#issuecomment-3061287887_

--- a/forge/forge/module.py
+++ b/forge/forge/module.py
@@ -349,7 +349,7 @@ class TFModule(Module):
     def get_parameters(self) -> List[Parameter]:
         params = []
         vars_as_pt = to_pt_tensors(self.module.trainable_variables)
-        names = [var.name for var in self.module.trainable_variables]
+        names = [var.path for var in self.module.trainable_variables]
         for param, name in zip(vars_as_pt, names):
             forge_param = Parameter(param, requires_grad=True, name=name)
             params.append(forge_param)

--- a/forge/forge/python_codegen.py
+++ b/forge/forge/python_codegen.py
@@ -426,7 +426,7 @@ class ForgeWriter(PythonWriter):
 
             self.wl("for weight in weights:")
             self.indent += 1
-            self.wl("name = weight.name")
+            self.wl("name = weight.path")
             if self.contains_incompatible_np_floats:
                 self.wl(
                     "# Some floating-point weights in the model havea a dtype that is incompatible with the .numpy() call."

--- a/forge/forge/tvm_calls/forge_compile.py
+++ b/forge/forge/tvm_calls/forge_compile.py
@@ -1285,13 +1285,13 @@ def format_tvm_graph_weights(inputs, module, compiler_cfg, framework=None):
         }
 
     elif framework == "tensorflow":
-        trainable_weight_names = [t_weight.name for t_weight in module.trainable_weights]
+        trainable_weight_names = [t_weight.path for t_weight in module.trainable_weights]
         weights = {
-            weight.name: (
+            weight.path: (
                 torch.Tensor(
                     tf.cast(weight, tf.float32).numpy() if tf.as_dtype(weight.dtype).is_floating else weight.numpy()
                 ),
-                weight.name in trainable_weight_names,
+                weight.path in trainable_weight_names,
             )
             for weight in module.weights
         }

--- a/forge/forge/tvm_calls/forge_utils.py
+++ b/forge/forge/tvm_calls/forge_utils.py
@@ -231,10 +231,10 @@ def construct_tvm_ir(framework: str, model, tvm_mod, params, compiler_cfg: Compi
         for (bad_name, value) in params.items():
             weight_found = False
             for tf_weight in model.weights:
-                if np.array_equal(tf_weight.numpy(), value.numpy()) and tf_weight.name not in found_weights:
-                    param_name_lookup[bad_name] = tf_weight.name
+                if np.array_equal(tf_weight.numpy(), value.numpy()) and tf_weight.path not in found_weights:
+                    param_name_lookup[bad_name] = tf_weight.path
                     weight_found = True
-                    found_weights.append(tf_weight.name)
+                    found_weights.append(tf_weight.path)
                     break
             if not weight_found:
                 param_name_lookup[bad_name] = bad_name

--- a/forge/test/mlir/test_ops_tf.py
+++ b/forge/test/mlir/test_ops_tf.py
@@ -95,7 +95,7 @@ def test_conv2d(
     verify(inputs, framework_model, compiled_model)
 
 
-@pytest.mark.xfail
+@pytest.mark.push
 def test_dual_conv2d():
 
     tf.random.set_seed(0)

--- a/forge/test/models/tensorflow/vision/resnet/test_resnet.py
+++ b/forge/test/models/tensorflow/vision/resnet/test_resnet.py
@@ -14,7 +14,7 @@ from forge.verify.verify import verify
 from test.models.tensorflow.vision.resnet.model_utils.image_utils import get_sample_inputs
 
 
-@pytest.mark.xfail
+@pytest.mark.push
 @pytest.mark.nightly
 def test_resnet_tensorflow():
 

